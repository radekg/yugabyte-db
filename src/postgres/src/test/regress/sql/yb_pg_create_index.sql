--
-- CREATE_INDEX
-- Create ancillary data structures (i.e. indices)
--

--
-- LSM
--
CREATE INDEX onek_unique1 ON onek USING lsm(unique1 int4_ops);

CREATE INDEX IF NOT EXISTS onek_unique1 ON onek USING lsm(unique1 int4_ops);

CREATE INDEX IF NOT EXISTS ON onek USING lsm(unique1 int4_ops);

CREATE INDEX onek_unique2 ON onek USING lsm(unique2 int4_ops);

CREATE INDEX onek_hundred ON onek USING lsm(hundred int4_ops);

CREATE INDEX onek_stringu1 ON onek USING lsm(stringu1 name_ops);

CREATE INDEX tenk1_unique1 ON tenk1 USING lsm(unique1 int4_ops);

CREATE INDEX tenk1_unique2 ON tenk1 USING lsm(unique2 int4_ops);

CREATE INDEX tenk1_hundred ON tenk1 USING lsm(hundred int4_ops);

CREATE INDEX tenk1_thous_tenthous ON tenk1 (thousand, tenthous);

CREATE INDEX tenk2_unique1 ON tenk2 USING lsm(unique1 int4_ops);

CREATE INDEX tenk2_unique2 ON tenk2 USING lsm(unique2 int4_ops);

CREATE INDEX tenk2_hundred ON tenk2 USING lsm(hundred int4_ops);

CREATE INDEX rix ON road USING lsm (name text_ops);

CREATE INDEX iix ON ihighway USING lsm (name text_ops);

CREATE INDEX six ON shighway USING lsm (name text_ops);

--
-- Try some concurrent index builds
--
-- Unfortunately this only tests about half the code paths because there are
-- no concurrent updates happening to the table at the same time.

CREATE TABLE concur_heap (f1 text, f2 text);
-- empty table
CREATE INDEX CONCURRENTLY concur_index1 ON concur_heap(f2,f1);

-- You can't do a concurrent index build in a transaction
BEGIN;
CREATE INDEX CONCURRENTLY concur_index7 ON concur_heap(f1);
COMMIT;

-- But you can do a regular index build in a transaction
BEGIN;
CREATE INDEX std_index on concur_heap(f2);
COMMIT;
