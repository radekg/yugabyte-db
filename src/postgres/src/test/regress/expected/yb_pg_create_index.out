--
-- CREATE_INDEX
-- Create ancillary data structures (i.e. indices)
--
--
-- LSM
--
CREATE INDEX onek_unique1 ON onek USING lsm(unique1 int4_ops);
CREATE INDEX IF NOT EXISTS onek_unique1 ON onek USING lsm(unique1 int4_ops);
NOTICE:  relation "onek_unique1" already exists, skipping
CREATE INDEX IF NOT EXISTS ON onek USING lsm(unique1 int4_ops);
ERROR:  syntax error at or near "ON"
LINE 1: CREATE INDEX IF NOT EXISTS ON onek USING lsm(unique1 int4_op...
                                   ^
CREATE INDEX onek_unique2 ON onek USING lsm(unique2 int4_ops);
CREATE INDEX onek_hundred ON onek USING lsm(hundred int4_ops);
CREATE INDEX onek_stringu1 ON onek USING lsm(stringu1 name_ops);
CREATE INDEX tenk1_unique1 ON tenk1 USING lsm(unique1 int4_ops);
CREATE INDEX tenk1_unique2 ON tenk1 USING lsm(unique2 int4_ops);
CREATE INDEX tenk1_hundred ON tenk1 USING lsm(hundred int4_ops);
CREATE INDEX tenk1_thous_tenthous ON tenk1 (thousand, tenthous);
CREATE INDEX tenk2_unique1 ON tenk2 USING lsm(unique1 int4_ops);
CREATE INDEX tenk2_unique2 ON tenk2 USING lsm(unique2 int4_ops);
CREATE INDEX tenk2_hundred ON tenk2 USING lsm(hundred int4_ops);
CREATE INDEX rix ON road USING lsm (name text_ops);
CREATE INDEX iix ON ihighway USING lsm (name text_ops);
ERROR:  relation "ihighway" does not exist
CREATE INDEX six ON shighway USING lsm (name text_ops);
ERROR:  relation "shighway" does not exist
--
-- Try some concurrent index builds
--
-- Unfortunately this only tests about half the code paths because there are
-- no concurrent updates happening to the table at the same time.
CREATE TABLE concur_heap (f1 text, f2 text);
-- empty table
CREATE INDEX CONCURRENTLY concur_index1 ON concur_heap(f2,f1);
ERROR:  CREATE INDEX CONCURRENTLY not supported yet
LINE 1: CREATE INDEX CONCURRENTLY concur_index1 ON concur_heap(f2,f1...
                     ^
HINT:  Please report the issue on https://github.com/YugaByte/yugabyte-db/issues
-- You can't do a concurrent index build in a transaction
BEGIN;
CREATE INDEX CONCURRENTLY concur_index7 ON concur_heap(f1);
ERROR:  CREATE INDEX CONCURRENTLY not supported yet
LINE 1: CREATE INDEX CONCURRENTLY concur_index7 ON concur_heap(f1);
                     ^
HINT:  Please report the issue on https://github.com/YugaByte/yugabyte-db/issues
COMMIT;
-- But you can do a regular index build in a transaction
BEGIN;
CREATE INDEX std_index on concur_heap(f2);
NOTICE:  making create index for table "concur_heap" nonconcurrent
DETAIL:  Create index in transaction block cannot be concurrent.
HINT:  Consider running it outside of a transaction block. See https://github.com/yugabyte/yugabyte-db/issues/6240.
COMMIT;
